src/uncertainties.jl

```diff
@@ -0,0 +1,41 @@
+## New Type: Quantity
+# Struct for carrying statistic and systematic uncertainty of a quantity
+struct Quantity{T<:Real} <: Real
+    val::T
+    stat_err::Tuple{T,T}
+    sys_err::Tuple{T,T}
+end
+
+# Main constructor (not exported)
+function _Quantity(val::T, err::Union{U,Tuple{U,V}}; err2::Union{W,Tuple{W,Q}}, type::S = "stat") where {T<:Real,U,V,W,Q,S<:AbstractString}
+    if contains(type, "stat")
+        Quantity(val, _to_tuple(T, err), _to_tuple(T, err2))
+    elseif contains(type, "sys")
+        Quantity(val, _to_tuple(T, err2), _to_tuple(T, err))
+    else
+        @error ArgumentError("The stated type of error '$type' is not recognized!\nOnly statistical and systematic errors are supported.")
+    end
+end
+
+# Ordered Constructor
+Quantity(val::T, stat_err::Union{U,Tuple{U,V}}, sys_err::Union{W,Tuple{W,X}} = zero(T)) where {T,U<:Real,V<:Real,W<:Real,X<:Real} = _Quantity(val, stat_err, err2 = sys_err)
+
+# Type-Named Constructor for tuple error ((a)symmetric)
+Quantity(val::T, err::Union{U,Tuple{U,V}}, type::S = "stat", err2::Union{W,Tuple{W,X}} = zero(T)) where {T,U,V,W<:Real,X<:Real,S<:AbstractString} = _Quantity(val, err, type = type, err2 = err2)
+
+# Special Constructors for tuple with error-type
+Quantity(val::T, err::Tuple{U,V,S}) where {T,U,V,S<:AbstractString} = Quantity(val, err[1:2], err[3])
+
+Quantity(val::T, err::Tuple{U,S}) where {T,U,S<:AbstractString} = Quantity(val, err[1], err[2])
+
+# Special Constructors
+const ±(val, err) = Quantity(val, err)
+const ±(Q1::Quantity{T}, err) where {T} = Q1 ± Quantity(zero(T), err)
+const ±(Q1::Quantity, Q2::Quantity) = Quantity(Q1.val + Q2.val, Q1.stat_err .+ Q2.stat_err, Q1.sys_err .+ Q2.sys_err)
+
+
+## Helper Functions to map single values and tuples to tuples of a specific type
+_to_tuple(x::T) where {T<:Real} = (x, x)
+_to_tuple(x::Tuple{U,V}) where {U<:Real,V<:Real} = promote(x...)
+_to_tuple(t::Type{T}, x::Union{U,Tuple{U,V}}) where {T,U,V} = convert(Tuple{T,T},_to_tuple(x))
+_to_tuple(t::Type{T}, x::Union{U,Tuple{U,V}}) where {T<:Integer,U,V} = convert(Tuple{T,T},ceil.(_to_tuple(x)))
```

Comment:
> We could simply overload the constructor using multiple dispatch instead of using helper function.

The main benefit of the helper functions is that someone can pass without a problem for instance the following:
`Quantity(1.5,0.5,(0.1,0.2))` i.e. a mixture of signatures and we don't need to do an overload for all possible combinations of this. Also they already handle the correct rounding properties if you pass an `Int` value and `Float` errors correspondingly. 
But if you insist on getting rid of helpers it's fine for me too.

> As it is I find the code a bit hard to read. Also I think we should try to work without strings, and just rely on the order that we feed into the constructor.

I agree that it is hard to read now (which is why I wanted another opinion). My reasoning for using strings additionally is to provide an interface if someone has only a systematic uncertainty for some reason so it is not necessary to write `Quantity(1.0,0,0.1)` but instead either `Quantity(1.0,0.1,type="sys")` or `Quantity(1.0,(0.1,"sys"))` 
Another way to achieve this would be to restrict `Quantity` to two arguments only `Quantity(val,err)` and use for `err` a named tuple?

Reply:
For situations like these we can use keyword-arguments. I think the following handles all the cases we want and errors for all others and also provides default values if no errors are specified.
```julia

_to_tuple(x<:Complex) = (x,x)
_to_tuple(x::Tuple) = x
Quantity(x;sys=zero(x),stat=zero(x)) = Quantity(x,_to_tuple(sys),_to_tuple(stat))

```

Reply:
Looks good to me.
I'll have a look at it. 
Since we are including `Complex` too it should be sufficient to have `_to_tuple(x<:Number) = (x,x)`

Reply:
I completely forgot about `Number`. This is definitely the preferred solution.

Reply:
Had to edit a few more things but will commit an updated version.